src/middleware/cloudflare/index.ts

```diff
@@ -23,12 +23,31 @@ export function createCloudflareHandler(
   } = {}
 ) {
   return async function (request: Request): Promise<Response> {
+    // Handle CORS pre-flight request.
+    if (request.method === "OPTIONS") {
+      return new Response(null, {
+        headers: {
+          "Access-Control-Allow-Origin": "*",
+          "Access-Control-Allow-Methods": "*",
+          "Access-Control-Allow-Headers":
+            "Content-Type, User-Agent, Authorization",
+        },
+      });
+    }
```

Comment:
If it's not too much work, could you pull out this change into a separate PR? I also wonder if we should move this into the abstract [`handle-request.ts`](https://github.com/octokit/oauth-app.js/blob/ebb86a011e060df6113330edcbb11436be2dadd1/src/middleware/handle-request.ts) method and maybe add an option to the server whether we want to enable CORS or not?

If I recall correctly, we have to set `access-control-allow-credentials` to true when the `Authorization` header is set, and we cannot set `access-control-allow-origin` to `*`.

Ages ago I've built a cors proxy for this purpose which got some usage, but it's possible that browsers evolved since that time so things might have changed:

https://github.com/gr2m/hapi-cors-headers/blob/52325cb14f85aebfe293a14a00a32fba1b4fe485/index.js#L12-L27

Reply:
I think `access-control-allow-credentials` is not required here because all request sent from the browser has `.credentials` defaults to `same-origin` and we manually set Authorization` header by code:

> The **`Access-Control-Allow-Credentials`** response header tells browsers whether to expose the response to frontend JavaScript code when the request's credentials mode (`Request.credentials`) is `include`.

> `same-origin`: Send user credentials (cookies, basic http auth, etc..) if the URL is on the same origin as the calling script. **This is the default value.**

I’ve tested on Safari and Chrome, without a `access-control-allow-credentials: true` response header, `auth-oauth-client-user.js` can correctly receive the response without any error.

I can add `access-control-allow-origin` as an option when creating the middleware because the `*` wildcard may be unsafe in some circumstances. If user think that is fine, they can `createNodeMiddleware(app, {allowOrigin: "*"});`, but specifying a list of domains explicitly is better. **When `allowOrigin` is omitted, CORS is disabled by default.**

If you agree with me that `access-control-allow-credentials` is not necessary here and an `allowOrigin` option behaves as described is okay, please let me know and I will send another PR later.

Thanks.

Reply:
> I’ve tested on Safari and Chrome, without a `access-control-allow-credentials: true` response header, `auth-oauth-client-user.js` can correctly receive the response without any error.

just to be sure: did you test sending the requests from another domain then the one the OAuth App is running? If so then I'm okay with it as is.

Reply:
Yes, I tested from different domain.